parser.go

```diff
@@ -1 +1,13 @@
 package parser
+
+import (
+	"github.com/asyncapi/parser/hlsp"
+	"github.com/asyncapi/parser/models"
+)
+
+// Parse receives either a YAML or JSON AsyncAPI document, and tries to parse it.
+func Parse(yamlOrJSONDocument []byte) (*models.ParsedAsyncAPI, *hlsp.ParserError) {
```

Comment:
I'm not getting why you are introducing this new struct ParsedAsyncAPI. can you give more info?

Reply:
I this to create a wrapper for everything that will be beautified on top of the original object?

Reply:
The ParsedAsyncAPI struct is the output of the parser. It will contain data from the beautification process but will also contain stuff from the rest of the modules (schema parsers and extension parsers). The fact that right now only contains things from the beautification process is just coincidence, because we haven't started with the other modules yet.

This is just separation of concerns. The beautification process belongs to the hlsp module, not to the parser as a whole. Remember the diagram in the plans.

Reply:
we could follow the same approach then we could do something like this. hlsp returning a beautified wrapper and this service returning another wrapper on top if the beautified struct like: ParsedObj -> BeatufiedObj -> AsyncAPI

Reply:
I'm not sure I understand what you're proposing.

Reply:
we can discuss this later, never mind.